.github/workflows/build-and-test.yml

```diff
@@ -104,19 +104,6 @@ jobs:
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Configure pact
```

Comment:
I'm moving the pact verification step(s) from the `build-and-test` action so that they only run on merges to main.